System.Server/Services/OrderService.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System.Server.Data;
+using System.Server.IServices;
+using System.Server.Models;
+using System.Server.Models.DTO;
+using System.Server.Enums;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Conventions;
+
+namespace System.Server.Services
+{
+    public class OrderService : IOrderService
+    {
+
+        private readonly SystemContext _context;
+        public OrderService(SystemContext context)
+        { 
+            _context = context; 
+        }
+
+        public async Task<IEnumerable<OrderGetDTO>> GetAllOrders()
+        {
+            var orders = await _context.Orders.ToListAsync();
+            var orderList = new List<OrderGetDTO>();
+            var n = orders.Count;
+            for (int i = 0; i < n; i++)
+            {
+                orderList.Add(await GetOrderById(orders[i].Id));
```

Comment:
Kodel vel viska pereit per duomenu baze, ar jau anksciau kviestas _context.Orders.ToListAsync() iskart negrazina to Task<IEnumerable<Order>>?

Vietoj to, kad kviest dar viena serviso metoda, kad sudet i DTO geriau manau butu tiesiog kazkokia OrderMapper klase kuri turetu statinius metodus, kurie paima kaip parametrais order ir galimai IEnumerable<orderProduct> ir grazina order DTO. 

Tuo pačiu manau sokinet tarp serviso metodu be reikalo nera labai gerai, pasistenk to irgi vengti

Reply:
sugalvosiu kažką